### ISSUE TITLE
Assertion error in 2016.1

### ISSUE THREAD

**Issue by cdegroot** [KronicDeth/intellij-elixir#262]
(I'm copy/pasting as I'm currently doing something else - may come back later and dive into the source code to see whether I can come up with a PR fix instead of a bug report, that would be my preferred way of contributing :-)). 

Having kafka_ex open in 2016.1 Ultimate:

```null
java.lang.AssertionError
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.operator(ElixirPsiImplUtil.java:929)
	at org.elixir_lang.psi.impl.ElixirUnmatchedMatchOperationImpl.operator(ElixirUnmatchedMatchOperationImpl.java:78)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.functionNameElement(ElixirPsiImplUtil.java:1660)
	at org.elixir_lang.psi.impl.ElixirUnmatchedMatchOperationImpl.functionNameElement(ElixirUnmatchedMatchOperationImpl.java:46)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.functionName(ElixirPsiImplUtil.java:1625)
	at org.elixir_lang.psi.impl.ElixirUnmatchedMatchOperationImpl.functionName(ElixirUnmatchedMatchOperationImpl.java:41)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.resolvedFunctionName(ElixirPsiImplUtil.java:3930)
	at org.elixir_lang.psi.impl.ElixirUnmatchedMatchOperationImpl.resolvedFunctionName(ElixirUnmatchedMatchOperationImpl.java:108)
	at org.elixir_lang.annonator.Kernel$1.visitCall(Kernel.java:185)
	at org.elixir_lang.annonator.Kernel$1.visitElement(Kernel.java:207)
	at com.intellij.psi.impl.PsiElementBase.accept(PsiElementBase.java:274)
	at org.elixir_lang.psi.impl.ElixirUnmatchedExpressionImpl.accept(ElixirUnmatchedExpressionImpl.java:19)
	at org.elixir_lang.psi.impl.ElixirUnmatchedMatchOperationImpl.accept(ElixirUnmatchedMatchOperationImpl.java:24)
	at org.elixir_lang.annonator.Kernel.annotate(Kernel.java:179)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.a(DefaultHighlightVisitor.java:135)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.visit(DefaultHighlightVisitor.java:99)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:351)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.access$100(GeneralHighlightingPass.java:62)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass$3.run(GeneralHighlightingPass.java:280)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:305)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.access$200(GeneralHighlightingPass.java:62)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass$4.run(GeneralHighlightingPass.java:311)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.analyze(DefaultHighlightVisitor.java:83)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:308)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.a(GeneralHighlightingPass.java:277)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectInformationWithProgress(GeneralHighlightingPass.java:216)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:82)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:70)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass$1$1.run(PassExecutorService.java:444)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1178)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass$1.run(PassExecutorService.java:435)
	at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:446)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:392)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.a(PassExecutorService.java:432)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:408)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:206)
	at jsr166e.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at jsr166e.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:858)
	at jsr166e.ForkJoinPool.scan(ForkJoinPool.java:1687)
	at jsr166e.ForkJoinPool.runWorker(ForkJoinPool.java:1642)
	at jsr166e.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:108)
```

and "occurred 14 times since the last clear", the counter keeps increasing. Could be that it trips over some halfbaked code-in-progress, as I said, I'll try to revisit later.

**Comment by KronicDeth** [comment]
It's due to an interaction of an precondition assertion in `Operation#operator` that assumes an `Operation` will always have 3 elements (left operand, operator, and right operand) and the automatic error recovery that the Pratt Parser in GrammarKit gives the Elixir parser, which leads to that precondition not being true.

This is a duplicate of #251, which was fixed in #259, which will be released in v2.3.0 as soon as I get the build matrix for #250 working.

**Comment by cdegroot** [comment]
Thanks for the quick follow-up, I think you saved me some time here :-)
